widgets/src/command_text_input.rs

```diff
@@ -0,0 +1,610 @@
+use crate::*;
+
+live_design! {
+    link widgets;
+    use link::widgets::*;
+    use link::theme::*;
+
+    List = {{List}} {
+        flow: Down,
+        width: Fill,
+        height: Fill,
+    }
+
+    pub CommandTextInput = {{CommandTextInput}} {
+        flow: Down,
+        height: Fit,
+
+        keyboard_focus_color: (THEME_COLOR_CTRL_HOVER),
+        pointer_hover_color: (THEME_COLOR_CTRL_HOVER * 0.85),
+
+        popup = <RoundedView> {
+            flow: Down,
+            height: Fit,
+            visible: false,
+            search_input = <TextInput> {
+                width: Fill,
+                height: Fit,
+            }
+
+            list = <List> {
+                height: Fit
+            }
+        }
+
+        persistent = <RoundedView> {
```

Comment:
Adding a couple of empty View slots here to simplify placement of stuff around the main text input. Without this, Moly's layout would be complex to recreate.